Orange/widgets/evaluate/owparameterfitter.py

```diff
@@ -318,93 +368,150 @@ def _add_plot(self):
-
-        gui.appendRadioButton(buttons, "Range")
-        hbox = gui.indentedBox(buttons, 20, orientation=Qt.Horizontal)
-        kw = {"minv": -MIN_MAX_SPIN, "maxv": MIN_MAX_SPIN,
-              "callback": self.__on_setting_changed}
-        self.__spin_min = gui.spin(hbox, self, "minimum", label="Min:", **kw)
-        self.__spin_max = gui.spin(hbox, self, "maximum", label="Max:", **kw)
-
-        gui.appendRadioButton(buttons, "Manual")
-        hbox = gui.indentedBox(box, 20, orientation=Qt.Horizontal)
-        gui.lineEdit(hbox, self, "manual_steps", placeholderText="10, 20, 30",
-                     callback=self.__on_setting_changed)
-
-        box = gui.vBox(self.controlArea, "Steps preview")
-        self.preview = ""
-        gui.label(box, self, "%(preview)s", wordWrap=True)
+        layout.addWidget(self.__combo, 0, 0, 1, 2)
+
+        buttons = gui.radioButtons(None, self, "type",
+                                   callback=self.__on_type_changed)
+        button = gui.appendRadioButton(buttons, "Range:")
+        layout.addWidget(button, 1, 0)
+
+        kw = dict(minv=-MIN_MAX_SPIN, maxv=MIN_MAX_SPIN,
+                  alignment=Qt.AlignRight,
+                  callback=self.__on_min_max_changed)
+        box = gui.hBox(None)
+        self.__spin_min = gui.spin(box, self, "minimum", label="From:", **kw)
+        layout.addWidget(box, 1, 1)
+
+        box = gui.hBox(None)
+        self.__spin_max = gui.spin(box, self, "maximum", label="To:", **kw)
+        layout.addWidget(box, 2, 1)
+
+        self.range_preview = RangePreview()
+        layout.addWidget(self.range_preview, 3, 0, 1, 2)
+
+        gui.appendRadioButton(buttons, "Manual:")
+        layout.addWidget(buttons, 4, 0)
+        edit = gui.lineEdit(None, self, "manual_steps",
+                            placeholderText="10, 20, 30",
+                            callback=self.__on_manual_changed)
+        layout.addWidget(edit, 4, 1)
+
+        # gui.lineEdit's connect does not call the callback on return pressed
+        # if the line hasn't changed.
+        @edit.returnPressed.connect
+        def _():
+            self.type = self.MANUAL
 
         gui.rubber(self.controlArea)
 
         gui.auto_apply(self.buttonsArea, self, "auto_commit")
 
+    def __on_type_changed(self):
+        self._settings_changed()
+
     def __on_parameter_changed(self):
         self._set_range_controls()
-        self.__on_setting_changed()
+        self._settings_changed()
+
+    def __on_min_max_changed(self):
+        self.type = self.FROM_RANGE
+        self._settings_changed()
 
-    def __on_setting_changed(self):
+    def __on_manual_changed(self):
+        self.type = self.MANUAL
+        self._settings_changed()
+
+    def _settings_changed(self):
         self._update_preview()
         self.commit.deferred()
 
     @property
     def fitted_parameters(self) -> list:
-        if not self._learner or not self._data:
+        if not self._learner:
```

Comment:
I refactored this because `self._data` is not needed if `self._learner` is not an instance of fitting. If this function can work without data, let it. The computation should be (and can be and is) stopped elsewhere.

The reason I like this better is because we may want to output a `Learner` from this widget. (That is, a Learner that has a base learner and performs fitting of arguments internally.) In this case, the widget will also have an active output without data.

Revert if you disagree. :)